arch/common/sw_isr_common.c

```diff
@@ -71,40 +71,28 @@ unsigned int get_parent_offset(unsigned int parent_irq,
 
 #endif /* CONFIG_MULTI_LEVEL_INTERRUPTS */
 
-void z_isr_install(unsigned int irq, void (*routine)(const void *),
-		   const void *param)
+unsigned int get_sw_isr_table_idx(unsigned int irq)
 {
 	unsigned int table_idx;
 
-	/*
-	 * Do not assert on the IRQ enable status for ARM GIC since the SGI
-	 * type interrupts are always enabled and attempting to install an ISR
-	 * for them will cause the assertion to fail.
-	 */
-#ifndef CONFIG_GIC
-	__ASSERT(!irq_is_enabled(irq), "IRQ %d is enabled", irq);
-#endif /* !CONFIG_GIC */
-
 #ifdef CONFIG_MULTI_LEVEL_INTERRUPTS
-	unsigned int level;
-	unsigned int parent_irq;
-	unsigned int parent_offset;
+	unsigned int level, parent_irq, parent_offset;
 
 	level = irq_get_level(irq);
 
 	if (level == 2U) {
 		parent_irq = irq_parent_level_2(irq);
 		parent_offset = get_parent_offset(parent_irq,
-			lvl2_irq_list,
-			CONFIG_NUM_2ND_LEVEL_AGGREGATORS);
+						  lvl2_irq_list,
+						  CONFIG_NUM_2ND_LEVEL_AGGREGATORS);
```

Comment:
should try to reduce formatting changes for clarity

Reply:
This is part of the change borrowed from #61422, will be rebased when that PR is merge